File: app/src/main/java/omniex/nl/omniex/ui/app/main/home/HomeFragment.java
package omniex.nl.omniex.ui.app.main.home;

import android.support.v7.widget.RecyclerView;

import org.androidannotations.annotations.AfterViews;
import org.androidannotations.annotations.EFragment;
import org.androidannotations.annotations.ViewById;

import java.util.List;

import javax.inject.Inject;

import nl.omniex.omniexshopping.R;
import nl.omniex.omniexshopping.data.model.products.FeaturedProduct;
import nl.omniex.omniexshopping.ui.adapters.FeatureProductsAdapter;
import nl.omniex.omniexshopping.ui.app.product.ProductDetailsFragment_;
import nl.omniex.omniexshopping.ui.base.BaseFragment;
import nl.omniex.omniexshopping.ui.base.BaseRecyclerAdapter;

@EFragment(R.layout.fragment_home)
public class HomeFragment extends BaseFragment<HomeView, HomePresenter> implements HomeView, BaseRecyclerAdapter.ItemClickListener<FeaturedProduct> {

    @ViewById(R.id.home_featured_products_rv)
    RecyclerView mFeaturedProductsRv;

    @Inject
    FeatureProductsAdapter mFeatureProductsAdapter;

    @AfterViews
    void initFeaturedProducts(){
        mFeatureProductsAdapter.setItemClickListener(this);
        mFeaturedProductsRv.setAdapter(mFeatureProductsAdapter);
        if(mFeatureProductsAdapter.getItemCount()==0)
        getPresenter().getListOfFeatured();
        getPresenter().getBestseller();
    }

    @Override
    public void onFeaturedProductsFetched(List<FeaturedProduct> featuredProducts) {
        mFeatureProductsAdapter.setItems(featuredProducts);
    }

    @Override
    public void onItemClick(FeaturedProduct featuredProduct) {
        goToFragment(ProductDetailsFragment_.builder().mProductId(featuredProduct.getProductId()).build(), true);
    }
}

File: app/src/main/java/omniex/nl/omniex/ui/app/main/home/HomeFragment.kt
package omniex.nl.omniex.ui.app.main.home

import android.support.v7.widget.RecyclerView
import omniex.nl.omniex.R
import omniex.nl.omniex.data.model.products.FeaturedProduct
import omniex.nl.omniex.ui.adapters.FeatureProductsAdapter
import omniex.nl.omniex.ui.app.product.ProductDetailsFragment_
import omniex.nl.omniex.ui.base.BaseFragment
import omniex.nl.omniex.ui.base.BaseRecyclerAdapter
import org.androidannotations.annotations.AfterViews
import org.androidannotations.annotations.EFragment
import org.androidannotations.annotations.ViewById
import javax.inject.Inject



@EFragment(R.layout.fragment_home)
open class HomeFragment : BaseFragment<HomeView, HomePresenter>(), HomeView, BaseRecyclerAdapter.ItemClickListener<FeaturedProduct> {

    @ViewById(R.id.home_featured_products_rv)
    lateinit var mFeaturedProductsRv: RecyclerView

    @Inject
    lateinit var mFeatureProductsAdapter: FeatureProductsAdapter

    @AfterViews
    internal fun initFeaturedProducts() {
        mFeatureProductsAdapter.setClickListener(this)
        mFeaturedProductsRv.adapter = mFeatureProductsAdapter
        if (mFeatureProductsAdapter.itemCount === 0)
            getPresenter().getListOfFeatured()
        getPresenter().getBestseller()
    }

    override fun onFeaturedProductsFetched(featuredProducts: List<FeaturedProduct>) {
        mFeatureProductsAdapter!!.setItems(featuredProducts)
    }

    override fun onItemClick(featuredProduct: FeaturedProduct?) {
        goToFragment(ProductDetailsFragment_.builder().mProductId(featuredProduct!!.productId).build(), true)
    }
}
